acceptance/suite_test.go

```diff
@@ -0,0 +1,261 @@
+	influx_client := influxdb.NewClient(influx_write_endpoint, "my-token")
+	write_api := influx_client.WriteAPIBlocking("my-org", "my-bucket")
+	s.api.influx_client = influx_client
+	s.api.writeAPI = write_api
+
+	// Prometheus client and API for verifying that writes occurred as expected
+	s.api.promClient, _ = promapi.NewClient(promapi.Config{
+		Address: fmt.Sprintf("http://%s:%s/api/prom", s.cfg.Docker.Host, s.cortexResource.GetPort("9009/tcp")),
+	})
+	s.api.querierClient = promv1.NewAPI(s.api.promClient)
+
+	s.suiteReady = time.Now()
+	s.T().Logf("Setup complete, took %s", time.Since(t0))
+}
+
+func (s *Suite) TearDownSuite() {
+	if s.T().Failed() {
+		if s.cfg.CI {
+			s.printLogs(s.influxProxyResource)
+		} else {
+			s.T().Logf("Not stopping containers to allow logs inspection")
+			s.T().Logf("Use the following command to stop them:")
+			s.T().Logf(`docker ps --filter "label=acceptance-suite=influx-proxy" -q | xargs docker stop | xargs docker rm`)
+			return
+		}
+	}
+
+	s.NoError(s.cortexResource.Close())
+	s.NoError(s.influxProxyResource.Close())
+	s.NoError(s.network.Close())
+}
+
+func (s *Suite) SetupTest() {
+	s.T().Logf("Test started at %s", time.Now().Format(time.RFC3339Nano))
+}
+
+func (s *Suite) TearDownTest() {
+	s.T().Logf("Test ended at %s", time.Now().Format(time.RFC3339Nano))
+}
+
+func (s *Suite) printLogs(res *dockertest.Resource) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	buf := &bytes.Buffer{}
+	s.Assert().NoError(s.pool.Client.Logs(docker.LogsOptions{
+		Context:           ctx,
+		Container:         res.Container.ID,
+		OutputStream:      buf,
+		InactivityTimeout: 0,
+		Stdout:            true,
+		Stderr:            true,
+		Timestamps:        true,
+		RawTerminal:       true, // put everything on OutputStream
+	}))
+
+	scanner := bufio.NewScanner(dlog.NewReader(buf))
+	s.T().Logf("Logs from %s", res.Container.Name)
+	for scanner.Scan() {
+		s.T().Logf("%s | %s", res.Container.Name, scanner.Text())
+	}
+}
+
+func (s *Suite) createNetwork() *dockertest.Network {
+	network, err := s.pool.CreateNetwork("acceptance-testing")
+	s.Require().NoError(err)
+	return network
+}
+
+func (s *Suite) startCortex() *dockertest.Resource {
+	const (
+		name = "cortex"
+		repo = "cortexproject/cortex"
+		tag  = "master-3018a54"
+	)
+
+	container := s.startContainer(&dockertest.RunOptions{
+		Name:         name,
+		Repository:   repo,
+		Tag:          tag,
+		Env:          nil,
+		Cmd:          []string{"cortex", "-config.file=/etc/config/cortex-config.yaml"},
+		Mounts:       []string{s.testFilePath() + "/config/cortex.yaml:/etc/config/cortex-config.yaml"},
+		ExposedPorts: []string{"9009"},
+		PortBindings: map[docker.Port][]docker.PortBinding{"9009/tcp": {}},
+		Networks:     []*dockertest.Network{s.network},
+		Labels:       suiteContainerLabels,
+	}, "ready", "9009/tcp")
+
+	return container
+}
+
+func (s *Suite) startInfluxProxy() *dockertest.Resource {
```

Comment:
continuing from above, would it be possible to move the influx-specific setup into a separate file so that we have a clean division between general acceptance test setup and this-proxy-specific stuff?